java/dataset/src/main/java/org/apache/arrow/dataset/jni/JniLoader.java

```diff
@@ -79,6 +79,7 @@ private void load(String name) {
     final String libraryToLoad = System.mapLibraryName(name);
     try {
       File temp = File.createTempFile("jnilib-", ".tmp", new File(System.getProperty("java.io.tmpdir")));
+      temp.deleteOnExit();
```

Comment:
Test: `mvn -Darrow.cpp.build.dir=/Users/arrow/lib -Dtest="TestFileSystemDataset#testBaseOrcRead" clean test`

Before the changes:
- target/jnilib-11079005003293013490.tmp
- target/jnilib-9915730910284592509.tmp

After the changes (1 tmp was deleted):
- target/jnilib-9915730910284592509.tmp

Could be possible to force delete also the another JNI file created?

Reply:
Sure, I can check and resolve this.

Reply:
Thank for your double check. I have remove the extra lib file and checked it in my environment.

Reply:
Awesome, thank you a lot